tests/Rules/UnusedPublicPropertyRule/Fixture/NullableProperty.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+namespace TomasVotruba\UnusedPublic\Tests\Rules\UnusedPublicPropertyRule\Fixture;
+
+use TomasVotruba\UnusedPublic\Tests\Rules\UnusedPublicPropertyRule\Source\PublicPropertyClass;
+
+class NullableProperty
+{
+    public function getBar(): ?PublicPropertyClass
+    {
+        if (rand(0, 1)) {
+            return null;
+        }
+        return new PublicPropertyClass();
+    }
+
+    public function doFoo(self $foo)
+    {
+        $x = $foo->getBar();
+        echo $x->property;
```

Comment:
unhandled case was here: $x is `null|PublicPropertyClass`, which the collector did not accept as a property fetch of `PublicPropertyClass->property`